src/gcp_scanner/scanner.py

```diff
@@ -166,7 +169,7 @@ def crawl_loop(initial_sa_tuples: List[Tuple[str, Credentials, List[str]]],
 
       project_id = project['projectId']
       project_number = project['projectNumber']
-      print(f'Inspecting project {project_id}')
+      logging.info(f'Inspecting project {project_id}')
```

Comment:
We actually want this to be printed in console so user understand what's going on.

Reply:
I think using logging for everything makes sense here and have the default log level be info. The output is metadata about the process, not the result of the process itself.

Reply:
> I think using logging for everything makes sense here and have the default log level be info. The output is metadata about the process, not the result of the process itself.

Will there always be a logging file as part of the arguments? coz I was thinking if that file name is optional and not mandatory we can separate the logging as a filehandle for file logging and console handler for console printing

something like this

```
import os
import logging

def main():
    log_level = getattr(logging, 'INFO', None)
    log_format = '%(asctime)s - %(levelname)s - %(message)s'
    date_format = '%Y-%m-%d %H:%M:%S'
    
    # Create a formatter
    formatter = logging.Formatter(fmt=log_format, datefmt=date_format)

    # Get the root logger
    logger = logging.getLogger()
    logger.setLevel(log_level)

    # Check if the log file exists
    if os.path.exists('my_log_file.log'):
        # If the log file exists, create a handler for the log file
        file_handler = logging.FileHandler(filename='my_log_file.log', mode='a')
        file_handler.setFormatter(formatter)
        logger.addHandler(file_handler)
    else:
        # If the log file does not exist, create a handler for the console
        console_handler = logging.StreamHandler()
        console_handler.setFormatter(formatter)
        logger.addHandler(console_handler)

    logger.info("this is a new line")

main()  # Call the main function
```

Reply:
> I think using logging for everything makes sense here and have the default log level be info. The output is metadata about the process, not the result of the process itself.

Yea, probably that's how it should be. I was thinking about having INFO to be default but in some cases it is just a lot of data in terminal. IMO, it would be great to allow user to specify verbosity level instead of just ERROR, WARNING, INFO, with default (verbosity 0 or 1) to be minimum required to understand that scanner is working and scanning projects (probably print now) and maximum (e.g. 3) where we print everything.

Reply:
> > I think using logging for everything makes sense here and have the default log level be info. The output is metadata about the process, not the result of the process itself.
> 
> Will there always be a logging file as part of the arguments? coz I was thinking if that file name is optional and not mandatory we can separate the logging as a filehandle for file logging and console handler for console printing
> 
> something like this
> 
> ```
> import os
> import logging
> 
> def main():
>     log_level = getattr(logging, 'INFO', None)
>     log_format = '%(asctime)s - %(levelname)s - %(message)s'
>     date_format = '%Y-%m-%d %H:%M:%S'
>     
>     # Create a formatter
>     formatter = logging.Formatter(fmt=log_format, datefmt=date_format)
> 
>     # Get the root logger
>     logger = logging.getLogger()
>     logger.setLevel(log_level)
> 
>     # Check if the log file exists
>     if os.path.exists('my_log_file.log'):
>         # If the log file exists, create a handler for the log file
>         file_handler = logging.FileHandler(filename='my_log_file.log', mode='a')
>         file_handler.setFormatter(formatter)
>         logger.addHandler(file_handler)
>     else:
>         # If the log file does not exist, create a handler for the console
>         console_handler = logging.StreamHandler()
>         console_handler.setFormatter(formatter)
>         logger.addHandler(console_handler)
> 
>     logger.info("this is a new line")
> 
> main()  # Call the main function
> ```

The log filename argument is optional. If it is set, we print in the file and if not the argument is None and it is simply console printing as far as I remember.